pkg/sshagent/sshagent.go

```diff
@@ -83,8 +91,13 @@ func (a *AgentServer) Serve(processLabel string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	err = selinux.SetSocketLabel("")
-	if err != nil {
+	// Reset socket label.
+	if err := selinux.SetSocketLabel(""); err == nil {
+		// Unlock the thread only if the process label could be restored
+		// successfully.  Otherwise leave the thread locked and the Go runtime
+		// will terminate it once it returns to the threads pool.
+		runtime.UnlockOSThread()
```

Comment:
Why not unlock either way after the SetSocketLabel call?

Reply:
Yes sounds better to me, I was trying to make it more like how its was implemented in libpod but now I think this needs to be corrected in libpod as well.